### PR TITLE
fix: Removes the opinion of uppercase headers on tables

### PR DESCRIPTION
**Pull request by eboyer**
Closes sap/fundamental#1292

### CODE REVIEW

**Comment by joseegm** [comment]
Test Version for this PR was deployed

Built with commit c4a7da6d54922376ff084ea2258533e86e820001

https://deploy-preview-1370--fundamental.netlify.com

File: scss/components/table.scss [hunk 1]
```diff
@@ -65,7 +65,6 @@ $block: #{$fd-namespace}-table;
         @include fd-type("-1");
         @include fd-var-color("color", $fd-table-header-color, --fd-table-header-color);
         @include fd-var-color("background-color", $fd-table-header-background-color, --fd-table-header-background-color);
-        text-transform: uppercase;
```
Comment (greg-a-smith) [inline scss/components/table.scss]: I have no objection to the removal of the transform -- that worked.

Discussion topic here ... is it desired to have the table headers be a smaller font size than the base font size in the table (and on the page)?  Typically, headings and table headers (`<th>`) are usually emphasized more, not less.  Previously, even though the header font size was smaller, the emphasis was achieved by the uppercase text.

Comment (bcullman) [reply scss/components/table.scss]: I see that #1292 is reference, but it does not go into where the request to remove uppercase originated from. 

If #1292 originated with Fiori3 (which it *SHOULD* have) then I bet that spec addresses the font size issue that @greg-a-smith has brought up as well.

Comment (eboyer) [reply scss/components/table.scss]: @saad-mo @LeoT7508 ~ Do you happen to have a quick link to the reference for tables? In our current UI Kit Sketch file release we're presenting all caps on the table headers as well. Leo, maybe this is something we can update for the next Sketch file release as well?

Comment (droshev) [reply scss/components/table.scss]: The request comes from the latest design update for tables. I am sure @LeoT7508 and @jeannewalters can bring more info.

Comment (bcullman) [reply scss/components/table.scss]: @droshev shared the mocks. it does appear the headers are larger as well. 

That said,I think *all* components are getting some changes - table and others will receive changes.

This PR does resolve the narrow issue asked for here. 🚢

**Comment by jeannewalters** [comment]
In the GD master file, the table headers are initial caps rather than all upper case.

**Comment by LeoT7508** [comment]
@droshev 

The Idle table column headers are neutral 1, the sorted table column headers at neutral 2. Design sheets detailing this can be found here:

https://app.zeplin.io/project/5a8c458b6e231fc65bddbdfb/dashboard?seid=5aea1886fd8af67dae9ab1bb

**Comment by eboyer** [comment]
@LeoT7508 ~ I think we're specifically looking for confirmation on the font size for the table headers.

**Comment by LeoT7508** [comment]
We are using -1 Text size (12px) and text 4 for color.

**Comment by droshev** [comment]
@eboyer @LeoT7508 do you think we can merge(and release this)?

**Comment by LeoT7508** [comment]
@droshev As long as it matches what's in Zeplin yes.

**Pull request merged**